.github/workflows/publish.yml

```diff
@@ -0,0 +1,62 @@
+name: publish
+
+on:
+  push:
+    branches: [master]
+    tags: ['*']
+
+jobs:
+  publish-scm-1:
+    if: github.ref == 'refs/heads/master'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: tarantool/rocks.tarantool.org/github-action@master
+        with:
+          auth: ${{ secrets.ROCKS_AUTH }}
+          files: expirationd-scm-1.rockspec
+
+  publish-tag:
+    if: startsWith(github.ref, 'refs/tags/')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      # Create a rockspec for the release.
+      - run: printf '%s=%s\n' TAG "${GITHUB_REF##*/}" >> "${GITHUB_ENV}"
+      - run: sed -E
+          -e "s/branch = '.+'/tag = '${{ env.TAG }}'/g"
+          -e "s/version = '.+'/version = '${{ env.TAG }}-1'/g"
+          expirationd-scm-1.rockspec > expirationd-${{ env.TAG }}-1.rockspec
+
+      # Create a source tarball for the release (.src.rock).
+      #
+      # `tarantoolctl rocks pack <rockspec>` creates a source
+      # tarball. It speeds up
+      # `tarantoolctl rocks install <module_name> <version>` and
+      # frees it from dependency on git.
+      #
+      # Important: Don't confuse this command with
+      # `tarantoolctl rocks pack <module_name> [<version>]`, which
+      # creates a **binary** rock or .all.rock (see [1]). Don't
+      # upload a binary rock of a Lua/C module to
+      # rocks.tarantool.org. Lua/C modules are platform dependent.
```

Comment:
But it's not so for expirationd because it's written in pure Lua. 
In current case it's possible to upload .all.rock (as e.g. in [errors](https://github.com/tarantool/errors/blob/master/.github/workflows/publish.yml#L46)).

Reply:
Stole from the error repository. Also create tagged rockspec via `tarantoolctl rocks new_version --tag ${{ env.TAG }}` instead of sed code.

Reply:
The `new_version` produces working, but non quite readable result. I would prefer `sed`. See details [here](https://github.com/tarantool/smtp/pull/38#discussion_r621460012).

Reply:
I rewritten this comment (left causion re binary rocks anyway). I use `sed` for generating a release rockspec. So, we can use tarantool 1.10 and avoid usage of unsupported tarantool release. The resulting release rockspec looks better now, but when copy-pasting the `sed` code for other modules, we should pay attention to whether single or double quotes are used in the scm-1 rockspec. Not a silver bullet...